### PR TITLE
Make language views case-insensitive

### LINKED ISSUE #4373: [Regression] Language codes below TPs are case-sensitive
Case insensitivity in language codes has regressed when accessing translation projects as reported by our L10N managers.

While from the user perspective both `/pt_BR/` and `/pt_br/` work and point to the same location, accessing a translation project produces inconsistent results: `/pt_BR/<project_code>/` is accessible, whereas `/pt_br/<project_code>/` is not.
*(For the sake of completeness, the actual case for the language code in the server is `pt_BR`)*

This can be reproduced in EN's production server:
* https://translate.evernote.com/pt_BR/
* https://translate.evernote.com/pt_br/
* https://translate.evernote.com/pt_BR/corenote_evernote/
* https://translate.evernote.com/pt_br/corenote_evernote/ **(404)**

While the previous behavior can be reproduced in MLO:
* https://mozilla.locamotion.org/pt_BR/
* https://mozilla.locamotion.org/pt_br/
* https://mozilla.locamotion.org/pt_BR/terminology/
* https://mozilla.locamotion.org/pt_br/terminology/

Accessing resources under the lowercase language code also suffers from this issue.

### PR DESCRIPTION
**Pull request by phlax**
Fixes #4373

### CODE REVIEW

**Comment by phlax** [comment]
@julen would you check if this fixes the issue - ta

File: pootle/apps/pootle_project/models.py [hunk 1]
```diff
@@ -314,36 +300,31 @@ def resources(self):
         :cls:`~pootle_store.models.Store` resource paths available for
         this :cls:`~pootle_project.models.Project` across all languages.
         """
-        cache_key = make_method_key(self, 'resources', self.code)
+        from virtualfolder.models import VirtualFolderTreeItem
```
Comment (unho) [inline pootle/apps/pootle_project/models.py]: Is it possible to import this if the app is not installed?

File: pootle/apps/pootle_project/models.py [hunk 2]
```diff
@@ -314,36 +300,31 @@ def resources(self):
...
+                vfolder__is_public=True,
+                pootle_path__regex="^/[^/].*/%s/" % self.code)
+            if 'virtualfolder' in settings.INSTALLED_APPS
+            else [])
+        resources = sorted(
+            {"/".join(x.split("/")[3:])
+             for x
+             in (set(stores.values_list("pootle_path", flat=True))
+                 | set(dirs.values_list("pootle_path", flat=True))
+                 | set(vftis.values_list("pootle_path", flat=True)))},
```
Comment (unho) [inline pootle/apps/pootle_project/models.py]: Is it possible to run `values_list` on a list?

**Comment by unho** [comment]
lgtm

File: pootle/apps/pootle_project/models.py (hunk 1 above)
Comment (phlax) [inline pootle/apps/pootle_project/models.py]: yep, in the default pootle setup atm, yep

File: pootle/apps/pootle_project/models.py (hunk 2 above)
Comment (phlax) [inline pootle/apps/pootle_project/models.py]: good spot

**Comment by julen** [comment]
This didn't do the trick unfortunately (cf. https://github.com/translate/pootle/issues/4373#issuecomment-171920779).

**Pull request merged**